tests/api/test_order.py

```diff
@@ -2898,3 +2919,75 @@ def test_delete_invoice_no_permissions(user_api_client, orders):
+    invoice = Invoice.objects.create(order=orders[0])
+    number = "01/12/2020/TEST"
+    url = "http://www.example.com"
+    variables = {
+        "id": graphene.Node.to_global_id("Invoice", invoice.pk),
+        "number": number,
+        "url": url,
+    }
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    content = get_graphql_content(response)
+    invoice.refresh_from_db()
+    assert invoice.status == InvoiceStatus.READY
+    assert invoice.number == content["data"]["updateInvoice"]["invoice"]["number"]
+    assert invoice.url == content["data"]["updateInvoice"]["invoice"]["url"]
+
+
+def test_update_invoice_single_value(user_api_client, permission_manage_orders, orders):
+    number = "01/12/2020/TEST"
+    invoice = Invoice.objects.create(order=orders[0], number=number)
+    url = "http://www.example.com"
+    variables = {
+        "id": graphene.Node.to_global_id("Invoice", invoice.pk),
+        "url": url,
+    }
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    content = get_graphql_content(response)
+    invoice.refresh_from_db()
+    assert invoice.status == InvoiceStatus.READY
+    assert invoice.number == number
+    assert invoice.url == content["data"]["updateInvoice"]["invoice"]["url"]
+
+
+def test_update_invoice_missing_number(
+    user_api_client, permission_manage_orders, orders
+):
+    invoice = Invoice.objects.create(order=orders[0])
+    url = "http://www.example.com"
+    variables = {
+        "id": graphene.Node.to_global_id("Invoice", invoice.pk),
+        "url": url,
+    }
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    content = get_graphql_content(response)
+    invoice.refresh_from_db()
+    assert content["data"]["updateInvoice"]["errors"][0]["field"] == "invoice"
+    assert invoice.url is None
+    assert invoice.status == InvoiceStatus.PENDING
+
+
+def test_update_invoice_no_permissions(user_api_client, orders):
+    invoice = Invoice.objects.create(order=orders[0])
+    variables = {
+        "id": graphene.Node.to_global_id("Invoice", invoice.pk),
+        "number": "01/12/2020/TEST",
+        "url": "http://www.example.com",
+    }
+    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    assert_no_permission(response)
+
+
+def test_update_invoice_invalid_id(user_api_client, permission_manage_orders):
+    variables = {"id": "SW52b2ljZToxMzM3", "number": "01/12/2020/TEST"}
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    content = get_graphql_content(response)
+    assert content["data"]["updateInvoice"]["errors"][0]["field"] == "id"
```

Comment:
In a test with errors, we should compare error codes.

Reply:
Good point